client/context/broadcast.go

```diff
@@ -44,6 +67,10 @@ func (ctx CLIContext) BroadcastTxCommit(txBytes []byte) (sdk.TxResponse, error)
 
 	res, err := node.BroadcastTxCommit(txBytes)
 	if err != nil {
+		if res := CheckTendermintError(err); res != nil {
```

Comment:
We need to only return the error when it's not a matched Tendermint error, i.e:

```go
res := CheckTendermintError(err)
if res != nil {
	return *res, nil
}

return sdk.TxResponse{}, err
```

Reply:
I think this code is doing the same as you mentioned, but I could rename it to make it more obvious

Reply:
Let's say an error is returned and that error _is_ a matched Tendermint error, we want to return a valid `TxResponse` and no error, but the current code would return an error, which would not render a TxResponse to the client. I think...